custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,292 @@
+                            FixtureItemField(
+                                field_value=_id,
+                            )
+                        ]
+                    ),
+                    "doses_per_week": FieldList(
+                        field_list=[
+                            FixtureItemField(
+                                field_value=value,
+                            )
+                        ]
+                    )
+                },
+                item_attributes={},
+            )
+            data_item.save()
+            cls.data_items.append(data_item)
+
+    def tearDown(self):
+        delete_all_users()
+        super(TestAdherenceUpdater, self).tearDown()
+
+    @classmethod
+    def tearDownClass(cls):
+        for domain in Domain.get_all():
+            domain.delete()
+        cls.data_type.delete()
+        for data_item in cls.data_items:
+            data_item.delete()
+        super(TestAdherenceUpdater, cls).tearDownClass()
+
+    def _create_episode_case(self, adherence_schedule_date_start, adherence_schedule_id):
+        person = get_person_case_structure(
+            self.person_id,
+            self.user.user_id,
+        )
+
+        occurrence = get_occurrence_case_structure(
+            self.occurrence_id,
+            person
+        )
+
+        episode_structure = get_episode_case_structure(
+            self.episode_id,
+            occurrence,
+            extra_update={
+                'adherence_schedule_date_start': adherence_schedule_date_start,
+                'adherence_schedule_id': adherence_schedule_id
+            }
+        )
+        cases = {case.case_id: case for case in self.factory.create_or_update_cases([episode_structure])}
+        return cases[self.episode_id]
+
+    def _create_adherence_cases(self, adherence_cases):
+        return self.factory.create_or_update_cases([
+            get_adherence_case_structure(
+                self.episode_id,
+                adherence_date,
+                extra_update={
+                    "name": adherence_date,
+                    "adherence_value": adherence_value,
+                }
+            )
+            for (adherence_date, adherence_value) in adherence_cases
+        ])
+
+    def assert_update(self, input, output):
+        update = self.calculate_adherence_update(input)
+        self.assertDictEqual(
+            update.update_json(),
+            output
+        )
+
+    def calculate_adherence_update(self, input):
+        self.case_updater.purge_date = pytz.UTC.localize(input[0])
+        # setup episode and adherence cases
+        adherence_schedule_date_start, adherence_schedule_id = input[1]
+        adherence_cases = input[2]
+        episode = self._create_episode_case(adherence_schedule_date_start, adherence_schedule_id)
+        self._create_adherence_cases(adherence_cases)
+
+        return EpisodeUpdate(episode, self.case_updater)
+
+    def test_adherence_schedule_date_start_late(self):
```

Comment:
@ctsims 
Hey Clayton, these are some test-cases for adherence update. Can you review these test cases and make sure they are upto spec?